ui/src/app/views/breadcrumb/component.ts

```diff
@@ -0,0 +1,111 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {ActivatedRoute, NavigationEnd, PRIMARY_OUTLET, Router} from '@angular/router';
+import {Subscription} from 'rxjs/Subscription';
+
+export interface Breadcrumb {
+  label: string;
+  url: string;
+}
+@Component({
+  selector: 'app-breadcrumb',
+  templateUrl: './component.html',
+  styleUrls: ['../../styles/buttons.css',
+    '../../styles/cards.css',
+    '../../styles/headers.css',
+    '../../styles/inputs.css',
+    './component.css']
+})
+export class BreadcrumbComponent implements OnInit, OnDestroy {
+  subscription: Subscription;
+  breadcrumbs: Breadcrumb[];
+  constructor(
+      private activatedRoute: ActivatedRoute,
+      private router: Router) {}
+
+  /**
+   * Generate a breadcrumb using the default label and url. Uses the route's
+   * paramMap to do any necessary variable replacement. For example, if we
+   * have a label value of ':wsid' as defined in a route's breadcrumb, we can
+   * do substitution with the 'wsid' value in the route's paramMap.
+   *
+   * @param {String} label
```

Comment:
nit: We're not using JS doc params elsewhere in the codebase AFAIK. I would probably not start adding them yet unless we decide to apply them everywhere.

I think they are generally less useful in Typescript than in normal JS, since you already have type information (e.g. in this case these annotations without comments aren't really adding anything).